packages/grid/_modules_/grid/hooks/features/infiniteLoader/useGridInfiniteLoader.ts

```diff
@@ -15,35 +15,38 @@ export const useGridInfiniteLoader = (apiRef: GridApiRef): void => {
   const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
   const isInScrollBottomArea = React.useRef<boolean>(false);
 
-  const handleGridScroll = React.useCallback(
-    (position: GridScrollParams) => {
-      if (!containerSizes || !position) {
-        return;
-      }
+  const getRealScroll = React.useCallback(
```

Comment:
that could is a reusable api method that this hook can reuse.
Maybe `getScrollPosition` as 'real' is not really meaningful, better placed in `useVirtualRows`

Reply:
And add it to the apiRef? I don't think it is a good idea as that would be 1 more API we need to support and that info is already accessible through the state.

Reply:
all infos are accessible through the state

Reply:
The goal is to provide a simple way of retrieving the info for users that don't know how to access the state.
If you want a low maintainance cost 

```tsx
const scrollPostionSelector = createSelector(
renderingSelector,
(renderingState)=> renderingState.realScroll //which can be renamed to something more meaningful. 
);

//now apiRef function is
getScrollPosition = useCallBack(()=> scrollPositionSelector(apiRef.current.getState()));
```

If there is a change you can just change the selector => no breaking changes for anyone

Reply:
How about I just use an existing selector -> `renderStateSelector(apiRef.current.getState());`
then we have 

```
const renderState = renderStateSelector(apiRef.current.getState());
renderState.realScroll.top
```

That way there is no need for any new selectors or APIs

Reply:
If we are going to have a selector why do you need also a method in the `apiRef`? The selectors are public.

Reply:
Well, it's a detail, more convenient for users to use apiRef than the selectors...
- ApiRef will have its documentation generated so it will be visible automatically in the api doc pages
- ApiRef.current would have the intellisense
- If you use a classComponent you won't use the `useGridSelector()`
- You have to know the selector exist as well as the state.

Reply:
> Well, it's more convenient for users to use apiRef than the selectors...
> 
> * ApiRef will have its documentation generated so it will be visible automatically in the api doc pages
> * ApiRef.current would have the intellisense
> * If you use a classComponent you won't use the `useGridSelector()`
> * You have to know the selector exist as well as the state.

Reply:
I can add it, it's fast, but I would like to hear also @oliviertassinari opinion on this because essentially if we are to do it here then we should do the same for other cases as well.

Reply:
Well it is the same (for most cases), check the row api, the column api, the selection, the sort, the filter...

Reply:
I personally think that we should keep everything private as long as no developers come up with a problem that exposing something would solve best than any other options.

Reply:
Updated

Reply:
@DanailH I guess it depends, if we have already data to support that this is required, then why not.